src/components/Ratings/Rating.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState, useEffect, FC } from "react";
+import StarIcon from "./StarIcon";
+import "./_Star.scss";
+
+export interface IRatingProps {
+    /**set true or false*/
+    disableHandler?: boolean;
+    /**set "sm" | "md" | "lg" */
+    starSize?: string;
+    /**set custom classname by sending in a prop*/
+    classNames?: string;
+    /**set star count by using array [1,2,3,4,5]*/
+    starCounts?: number[];
+    /**set customized default value of star. Example: 3*/
+    customRatingValue?: number;
+    /**set true or false to activate label on hover*/
+    hoverLabel?: boolean;
+    /**set custom icon size by sending prop in. Example: "75px" */
+    customSize?: string;
+    /**set label title by sending in prop. Example: "TESTING LABEL"*/
+    labelTitle?: string;
+    /**send addon-action from storybook as a prop */
+    onChange?: any
+}
+
+export type patRatingProps = IRatingProps;
+
+export const Rating: FC<patRatingProps> = (props: any) => {
+    const {
+        disableHandler,
+        starSize,
+        classNames,
+        starCounts,
+        customRatingValue,
+        hoverLabel,
+        customSize,
+        labelTitle,
+        customStarCount,
+        onChange
+    } = props;
+
+    const [rating, setRating] = useState<number>(customRatingValue);
+    const [hoverRating, setHoverRating] = useState<number>(0);
+    // const [starCounts, setstarCounts] = useState<number[]>(starCounts);
+    const [disableCss, setDisableCss] = useState<string>("");
+    const [ratingLabel, setRatingLabel] = useState<string>("");
+
+    const starLabel: any = {
```

Comment:
I really like what you did with labels here, but it would be better if you could allow users to customize labels. Maybe we could use prop, maybe something else. Also, I would like to see these labels applied to fraction of stars.

Reply:
ok! i am still having some trouble with the half-star logic so I will continue my research until I can find a solution.

Reply:
I added the custom hover label feature without the half star logic for now otherwise it works for the whole stars.